### DISCUSSION TITLE
Change status by request

### DISCUSSION THREAD

**Discussion by demetera** [krisppurg/dimscord#99] [Q&A] [answered]
When I call external function (when compiling):

proc statusOffline*(s: Shard) {.event(discord).} = 
  await s.updateStatus(activity = some ActivityStatus(name: "around.", kind: atPlaying), status = "invisible")

from another file, I get this : 
Error: in expression 'discord.events.statusOffline*': identifier expected, but found 'statusOffline*'

or updateStatus intended to run only from onReady() proc?

**Comment by krisppurg** [comment, marked as answer]
statusOffline is not an event, it can run through any event e.g. message create, guild create, ready. Ideally it would be better to set it to onReady proc.

You can ask questions on the channel at discord api server or dimscord server.
